serverlessworkflow/modules/ROOT/pages/getting-started/create-your-first-workflow-service-with-kn-cli-and-vscode.adoc

```diff
@@ -0,0 +1,101 @@
+= Creating and running workflow projects using KN CLI and Visual Studio Code
+
+This guide showcases the use of our Knative Workflow CLI plugin and Visual Studio code to create & run {product_name} projects.
+
+.Prerequisites
+* xref:tooling/kn-plugin-workflow-overview.adoc[Knative Workflow CLI] {kn_cli_version} is installed.
+* Visual Studio Code with https://marketplace.visualstudio.com/items?itemName=redhat.vscode-extension-serverless-workflow-editor[Red Hat Serverless Workflow Editor] is installed to edit your workflows.
+
+.Preparing your environment 
+* Install https://docs.docker.com/engine/install/[Docker] or https://podman.io/docs/installation[Podman].
+* Install https://minikube.sigs.k8s.io/docs/start/[minikube].
+* Install https://kubernetes.io/docs/tasks/tools/[kubectl].
+* Start minikube. Depending on your environment, set `--driver` flag to `podman` or `docker`
+[source,bash]
+----
+minikube start --cpus 4 --memory 8096 --addons registry --addons metrics-server --insecure-registry "10.0.0.0/24" --insecure-registry "localhost:5000"  --driver=docker
+----
+* (optional) Install https://k9scli.io/[k9scli.io] 
+* Install the https://operatorhub.io/operator/sonataflow-operator[Sonataflow Operator]
+* Install https://github.com/kiegroup/kie-tools/releases/tag/0.30.3[KN Workflow CLI] by downloading the correct distrubiton for your development environment and adding it to the PATH.
+
+[[proc-creating-app-with-kn-cli]]
+== Creating a workflow project with Visual Studio Code and KN CLI
+
+Use the `create`` command with kn workflow to scaffold a new Sonata Flow project.
+
+* Navigate to you development directory and create your project.
+[source,bash]
+----
+kn workflow create -n my-sonataflow-project
+----
+* This will create a folder with name `my-sonataflow-project` and a sample workflow `wokrflow.sw.json`
+[source,bash]
+----
+cd ./my-sonataflow-project
+----
+* Open the folder in Visual Studo Code and examine the created `worklow.sw.json` using our extension.
+
+Now we can run the project and execute your workflow.
+
+[[proc-running-app-with-kn-cli]]
+== Running a workflow project with Visual Studio Code and KN CLI
+
+Use the `run` command with kn workflow to build and run the Sonata Flow project in local development mode.
+
+* Run the project. This will start a `kogito-swf-devmode-nightly` image and map your local folder to this image.
+[source,bash]
+----
+kn workflow run
+----
+* The Development UI wil be accesible at `localhost:8080/q/dev`
+* You can now work on your project. Any changes will be picked up by the hot reload feature.
+* See xref:testing-and-troubleshooting/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-instances.adoc[Workflow instances] guide on how to run workflows via Development UI.
```

Comment:
Ins't this guide Quarkus specific?

Reply:
True, however the dev ui is accessible and when you use the CLI run command so I referenced this existing guide to avoid duplication, let me know if there is a better alternative to this. I can always remove this point and just point users to the URL.